File: 30-days-of-code/cpp/day-14-scope.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

class Difference {
    private:
    vector<int> elements;
  
  	public:
  	int maximumDifference;

    // Begin Solution
    //---------------------------------------------------------------------------------------
    Difference(vector<int> elements) {
        this->elements = elements;
    }

    void computeDifference() {
        int maxAbsolute = 0;
        int currDiff;
        for (int i = 0; i < this->elements.size(); i++) {
            for(int j = 0; j < this->elements.size(); j++) {
                currDiff = abs(elements[i] - elements[j]);
                if (currDiff > maxAbsolute) {
                    maxAbsolute = currDiff;
                }
            }
        }
        maximumDifference = maxAbsolute;
    }
    //---------------------------------------------------------------------------------------
    // End Solution
    
};

int main() {
    int N;
    cin >> N;
    
    vector<int> a;
    
    for (int i = 0; i < N; i++) {
        int e;
        cin >> e;
        
        a.push_back(e);
    }
    
    Difference d(a);
    
    d.computeDifference();
    
    cout << d.maximumDifference;
    
    return 0;
}

File: 30-days-of-code/java/day-14-scope.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;


class Difference {
  	private int[] elements;
  	public int maximumDifference;

	// Add your code here
    public Difference(int[] elements) {
            this.elements = elements;
    }
    void computeDifference() {
        int maxAbsolute = 0;
        int currDiff;
        for (int i = 0; i < this.elements.length; i++) {
            for(int j = 0; j < this.elements.length; j++) {
                currDiff = Math.abs(elements[i] - elements[j]);
                if (currDiff > maxAbsolute) {
                    maxAbsolute = currDiff;
                }
            }
        }
        maximumDifference = maxAbsolute;
    }
} // End of Difference class

public class Solution {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }
        sc.close();

        Difference difference = new Difference(a);

        difference.computeDifference();

        System.out.print(difference.maximumDifference);
    }
}